### DISCUSSION TITLE
Optimizing for online use with smartphone

### DISCUSSION THREAD

**Discussion by GridsUnitsPlanes** [textadventures/quest#1295] [Creating games with Quest]
I have made a few small text adventures but have noticed some quirks when played on an smartphone (online via link). I know that using a laptop/PC is more effective than a phone for viewing the games but would still like to optimize for smartphone (iOS and android) play (online) where possible. 

Specifically, text in the command bar does not appear when typing (I tried the text adventure from the link... using Safari... on iOS). 

I do tend to prefer either minimal layouts or retro layouts with dark backgrounds and light fonts. I also tend to use webfonts.

Note: I am not interested in developing an app, I am interested in optimising for both PC and smartphone play.

Any suggestions or tips for optimizing for smartphone play (online)?

**Comment by rheadkid** [comment]
> Specifically, text in the command bar does not appear when typing (I tried the text adventure from the link... using Safari... on iOS).

Hello. I've never heard anyone mention this before.

What link are you clicking, or which text adventure is it?

**Reply by GridsUnitsPlanes** [reply]
Cheers for the reply.

Here are three games I made that have this problem.

[https://textadventures.co.uk/games/view/-Xp0QeAMAU2_4NlqN7uGVg/diverge-and-coalesce](url)

[https://textadventures.co.uk/games/view/hunsf_zxvei6x7at1wii7g/on-as](url)

[https://textadventures.co.uk/games/view/vpmuszdpjust1xrwkyoguq/creole-aeon](url) - N.B. This was my first game and is buggy in other areas.

When you type in the command bar on a phone...no text appears. It still 'works' but it is tricky to know if you are spelling a phrase correctly.

**Reply by KVonGit** [reply]
Hi, I tested those on an Android phone, and *Diverge and Coalesce* is the only one that does that to me (not to say I don't believe you, I just don't have an iPhone here to test with, haha).

---
I think *Diverge and Coalesce* does that because the text input field's background color is black (actually transparent and the background behind it is black), and so is the text's color.

![image](https://github.com/user-attachments/assets/6510c83d-314c-45ac-915a-32dfb462eed3)

---
ON//AS

This one ... I don't know, but I do notice I can't read the text in the menus when I click the links either.

![image](https://github.com/user-attachments/assets/282704b7-3a96-4d83-bf89-db2458e67e91)

---
I can't replicate that in the third game on any of my devices.

---
This one is going to require someone with CSS skills. I thought I had a sloppy fix for one game, but it doesn't work when I use Firefox to emulate an iPhone.

**Reply by GridsUnitsPlanes** [reply]
Thank you. I will look into this and make sure I don't do this in future games. Thanks for this.
